src/components/common/FilterDropdownMenu.jsx

```diff
@@ -0,0 +1,28 @@
+import Form from 'react-bootstrap/Form';
+import React, { useState } from 'react';
+
+const CustomMenu = React.forwardRef(
+  ({ children, style, className, 'aria-labelledby': labeledBy }, ref) => {
+    const [value, setValue] = useState('');
+
+    return (
+      <div ref={ref} style={style} className={className} aria-labelledby={labeledBy}>
+        <Form.Control
+          autoFocus
+          className="mx-3 my-2 w-auto"
+          placeholder="Type to filter..."
+          onChange={(e) => setValue(e.target.value)}
+          value={value}
+        />
+        <ul className="list-unstyled">
+          {React.Children.toArray(children).filter(
+            (child) => !value || child.props.children.toLowerCase().startsWith(value.toLowerCase())
```

Comment:
Because there are sometimes schools like "The Ohio State University" and a user would probably just start the search with "Ohio"

```suggestion
            (child) => !value || child.props.children.toLowerCase().includes(value.toLowerCase())
```